ide/editor.completion/src/org/netbeans/spi/editor/completion/CompletionCollector.java

```diff
@@ -0,0 +1,565 @@
+            }
+
+            /**
+             * A string that should be used when filtering a set of completions.
+             * When {@code null} the label is used as the filter.
+             */
+            @NonNull
+            public Builder filterText(@NonNull String filterText) {
+                this.filterText = filterText;
+                return this;
+            }
+
+            /**
+             * A string that should be inserted into a document when selecting
+             * this completion. When {@code null} the label is used as the insert text.
+             */
+            @NonNull
+            public Builder insertText(@NonNull String insertText) {
+                this.insertText = insertText;
+                return this;
+            }
+
+            /**
+             * The format of the insert text. The format applies to both the
+             * {@code insertText} property and the {@code newText} property of a provided
+             * {@code textEdit}. If omitted defaults to {@link TextFormat#PlainText}.
+             */
+            @NonNull
+            public Builder insertTextFormat(@NonNull TextFormat insertTextFormat) {
+                this.insertTextFormat = insertTextFormat;
+                return this;
+            }
+
+            /**
+             * An edit which is applied to a document when selecting this completion.
+             * When an edit is provided the value of {@code insertText} is ignored.
+             * The range of the edit must be a single line range and it must
+             * contain the position at which completion has been requested.
+             */
+            @NonNull
+            public Builder textEdit(@NonNull TextEdit textEdit) {
+                this.textEdit = textEdit;
+                return this;
+            }
+
+            /**
+             * A list of additional text edits that are applied when selecting this
+             * completion. Edits must not overlap (including the same insert position)
+             * with the main edit nor with themselves.
+             * Additional text edits should be used to change text unrelated to the
+             * current cursor position (for example adding an import statement at the
+             * top of the file if the completion item will insert an unqualified type).
+             */
+            @NonNull
+            public Builder additionalTextEdits(@NonNull List<TextEdit> additionalTextEdits) {
+                this.additionalTextEdits = additionalTextEdits;
+                return this;
+            }
+
+            /**
+             * A list of characters that when pressed while this completion is
+             * active will accept it first and then type that character. All
+             * commit characters should have length one and that superfluous
+             * characters will be ignored.
+             */
+            @NonNull
+            public Builder commitCharacters(@NonNull List<String> commitCharacters) {
```

Comment:
Consider making this cumulative and change to parameter like `char...` or `char`.

Reply:
Fixed.